src/main/scala/spray/json/package.scala

```diff
@@ -32,7 +32,7 @@ package object json {
 
 package json {
 
-  class DeserializationException(msg: String, cause: Throwable = null) extends RuntimeException(msg, cause)
+  case class DeserializationException(msg: String, cause: Throwable = null, fieldName: Option[String] = None) extends RuntimeException(msg, cause)
```

Comment:
I agree that getting the type info would be another great thing to have.
However, while gathering up the field names is an easy, noninvasive thing we can do here, including the type info is a bit more difficult to do properly.